### ISSUE TITLE
Data from Kafka are not imported.

### ISSUE THREAD

**Issue by spyros87** [yandex/ClickHouse#5545]
I am using **Clickhouse 19.5.3.8** and when the Kafka topic lag increases, no more data are imported. For importing data, I need to reset the topic offset and then reimport them. As soon as the lag is small, CH starts importing data properly.

The **CH Kafka Consumer** settings are defined as follows:
```
Kafka SETTINGS kafka_broker_list = '192.168.54.48:9092',
               kafka_topic_list = 'topic',
               kafka_group_name = 'topic-group',
               kafka_format = 'JSONEachRow',
               kafka_num_consumers = 3,
               kafka_row_delimiter = '\n',
               kafka_skip_broken_messages = 1
```

These are the following Kafka Consumer Group settings:
```
TOPIC   PARTITION  CURRENT-OFFSET  LOG-END-OFFSET  LAG             CONSUMER-ID                                              HOST              CLIENT-ID
topic   1          32416794        39991572        7574778         ClickHouse 19.5.3.8-86393ea9-0eb5-4b0a-9e48-7366f494842f /192.168.54.48    ClickHouse 19.5.3.8
topic   2          32419989        39991572        7571583         ClickHouse 19.5.3.8-d756928d-b08c-4147-acd2-181aa962cbba /192.168.54.48    ClickHouse 19.5.3.8
topic   0          32420018        39991572        7571554         ClickHouse 19.5.3.8-5615e032-59ae-47e0-a860-8b0cd3f80536 /192.168.54.48    ClickHouse 19.5.3.8
```

**Comment by spyros87** [comment]
Maybe is related to #4631

**Comment by spyros87** [comment]
@CurtizJ This issue occurred on version 19.7.3.9 too.

**Comment by spyros87** [comment]
@CurtizJ I have made an observation regarding the issue. The documentation in EN mentions:

> kafka_skip_broken_messages – Kafka message parser mode. If kafka_skip_broken_messages = 1 then the engine skips the Kafka messages that can't be parsed (a message equals a row of data).

Which I have interpreted as a `boolean` value for skipping or not invalid messages. With the value set to 1, any parsing exception from [deeper levels](https://github.com/yandex/ClickHouse/blob/c846d0f6af1e8c835cddc8530360d09d41216f1b/dbms/src/Storages/Kafka/StorageKafka.cpp#L357) bubbled up to [`StorageKafka::streamThread()`](https://github.com/yandex/ClickHouse/blob/c846d0f6af1e8c835cddc8530360d09d41216f1b/dbms/src/Storages/Kafka/StorageKafka.cpp#L284). There, the exception is caught and logged and since the Stream is not canceled it won't reschedule itself in order to reconnect ([see here](https://github.com/yandex/ClickHouse/blob/c846d0f6af1e8c835cddc8530360d09d41216f1b/dbms/src/Storages/Kafka/StorageKafka.cpp#L310)). Since @vavrusa implemented this it might be the case that he is able to shed more light.

However by checking the code, I have realized that `kafka_skip_broken_messages` is of `size_t` type (aka unsigned int). With that in mind, the parameter `kafka_skip_broken_messages` is the tolerance to broken messages that might occur. Using this approach, I was able to import the data from Kafka to CH by setting a number big enough to consume the data.

If the `kafka_skip_broken_messages` number is reached then the connection to Kafka will be dropped. This might explain the behavior observed by [here](https://github.com/yandex/ClickHouse/issues/4631#issuecomment-501643367) and [here](https://github.com/yandex/ClickHouse/issues/4631#issuecomment-494685788) that after some time the symptoms return.

**Comment by vavrusa** [comment]
It looks like that's the intended behavior https://github.com/yandex/ClickHouse/commit/89ffb577b1a1cff5af8b98f48d3a44b810ee0807 and the documentation should not say it's like a boolean.

> There, the exception is caught and logged and since the Stream is not canceled it won't reschedule itself in order to reconnect (see here).

It's the other way round. It will reschedule itself UNLESS the stream is cancelled. The stream is cancelled when you drop or detach the table.

**Comment by abyss7** [comment]
If an issue still persists, then can you provide any related logs to see actual error messages?

**Comment by k0st1an** [comment]
I seem to have the same problem. Version 19.11. The consumer is connected (this is seen on the Kafka side), but the lag increases. At the same time on version 19.7.3 this problem is also there, but it is not so common. Data in the JSON topic. Tried to catch something in the logs, but there is silence.

**Comment by filimonov** [comment]
@k0st1an are you able to reproduce the problem with docker compose?

**Comment by k0st1an** [comment]
Play does not work. Nearby stood two versions of 19.7 and 19.11. 19.11 the problem were found much more often in almost all the topic. After rollback to version 19.7 no problems so far.

**Comment by abyss7** [comment]
Should be fixed in 19.14+. Possibly was backported to 19.13 and 19.11
